src/libs/data-utils.js

```diff
@@ -340,3 +355,254 @@ export const EntityUploader = class EntityUploader extends Component {
+            legend({ style: { marginBottom: '0.5rem' } }, [isList ? 'List item type:' : 'Attribute type:']),
+            h(Fragment, _.map(({ type, tooltip }) => h(TooltipTrigger, {
+              content: tooltip
+            }, [
+              span({ style: { marginRight: '1.2rem' } }, [h(RadioButton, {
+                text: _.startCase(type),
+                name: 'edit-type',
+                checked: editType === type,
+                onChange: () => {
+                  const convertFn = type === 'number' ?
+                    v => {
+                      const numberVal = _.toNumber(v)
+                      return _.isNaN(numberVal) ? 0 : numberVal
+                    } :
+                    Utils.convertValue(type === 'reference' ? 'string' : type)
+                  const convertedValue = isList ? _.map(convertFn, newValue) : convertFn(newValue)
+
+                  setNewValue(convertedValue)
+                  setEditType(type)
+                },
+                labelStyle: { paddingLeft: '0.5rem' }
+              })])
+            ]), [
+              { type: 'string' },
+              { type: 'reference', tooltip: 'A link to another entity' },
+              { type: 'number' },
+              { type: 'boolean' }
+            ]))
+          ]),
+          editType === 'reference' && div({ style: { marginTop: '0.5rem' } }, [
+            div({ style: { marginBottom: '0.5rem' } }, 'Referenced entity type:'),
+            h(Select, {
+              value: linkedEntityType,
+              options: entityTypes,
+              onChange: ({ value }) => setLinkedEntityType(value)
+            })
+          ])
+        ]),
+        div({ style: { marginBottom: '0.5rem' } }, [
+          h(LabeledCheckbox, {
+            checked: isList,
+            onChange: willBeList => setNewValue(willBeList ? [newValue] : newValue[0])
+          }, [span({ style: { marginLeft: '0.5rem' } }, ['Attribute is a list'])])
+        ]),
+        isList ?
+          div({ style: { marginTop: '1.5rem' } }, _.map(([i, value]) => div({
+            style: { display: 'flex', alignItems: 'center', marginBottom: '0.5rem' }
+          }, [
+            makeInput({
+              'aria-label': `List value ${i + 1}`,
+              autoFocus: true,
+              value,
+              onChange: v => setNewValue(_.set(i, v))
+            }),
+            h(Link, {
+              'aria-label': `Remove list value ${i + 1}`,
+              disabled: newValue.length === 1,
+              onClick: () => setNewValue(_.pullAt(i)),
+              style: { marginLeft: '0.5rem' }
+            }, [
+              icon('times', { size: 20 })
+            ])
+          ]), Utils.toIndexPairs(newValue))) :
+          div({ style: { marginTop: '1.5rem' } }, [
+            makeInput({
+              'aria-label': 'New value',
+              autoFocus: true,
+              value: newValue,
+              onChange: setNewValue
```

Comment:
This is throwing a console warning when changing booleans, because the `Switch` `onChange` sends back additional arguments after the value. We should either explicitly accept just the value here, or wrap `onChange` in our `Switch` do discard the extras.